### PR TITLE
LV-893: add scalac options for better code style

### PR DESCRIPTION
**Pull request by luchenyuxx**

### CODE REVIEW

File: src/test/scala/co/ledger/wallet/daemon/api/CurrenciesApiTest.scala [hunk 1]
```diff
@@ -69,19 +69,19 @@ class CurrenciesApiTest extends APIFeatureTest {
...
+//    1,
+//    "bitcoin",
+//    List(
+//      UnitView("testnet satoshi","satoshi","satoshi",0),
+//      UnitView("testnet bitcoin", "BTC", "BTC", 8),
+//      UnitView("testnet milli-bitcoin","mBTC", "mBTC", 5),
+//      UnitView("testnet micro-bitcoin", "μBTC", "μBTC", 2),
+//    ),
+//    BitcoinNetworkParamsView("btc_testnet", "6F", "C4", "043587CF", "PER_BYTE", 546, "Bitcoin signed message:\n", false)
+//  )
```
Comment (jchampemont) [inline src/test/scala/co/ledger/wallet/daemon/api/CurrenciesApiTest.scala]: If it's not used anymore I would delete it. It's still somewhere in git history.

image

Comment (aymeric-ledger) [reply src/test/scala/co/ledger/wallet/daemon/api/CurrenciesApiTest.scala]: 👍

File: src/main/scala/co/ledger/wallet/daemon/services/ECDSAService.scala [hunk 2]
```diff
@@ -22,5 +22,5 @@ class ECDSAService extends DaemonService {
 
   private def lease: Lease[Secp256k1] = _secp256k1Instances.acquire()
 
-  private lazy val _secp256k1Instances: Pool[Secp256k1] = Pool[Secp256k1](Runtime.getRuntime.availableProcessors(), Secp256k1.newInstance)
+  private lazy val _secp256k1Instances: Pool[Secp256k1] = Pool[Secp256k1](Runtime.getRuntime.availableProcessors(), () => Secp256k1.newInstance())
```
Comment (tt810) [inline src/main/scala/co/ledger/wallet/daemon/services/ECDSAService.scala]: why is this change?

Comment (luchenyuxx) [reply src/main/scala/co/ledger/wallet/daemon/services/ECDSAService.scala]: "Eta-expansion of zero-argument method values is deprecated. "
Given by the compiler.

**Pull request merged**